### PR TITLE
refactor: remove tokenlogger

### PR DESCRIPTION
**Pull request by liammoat**
## Purpose

* Removes the redundant implementation of `TokenLogger`. 

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix (`fix`)
[ ] Feature (`feat`)
[ ] Code style update (formatting, local variables) (`style`)
[X] Refactoring (no functional changes, no api changes) (`refactor`)
[ ] Documentation content changes (`docs`)
[ ] Other... Please describe: (`perf`, `test`, `build`, `ci`, `chore`, `revert`)
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Run the project, ensuring `APPINSIGHTS_ENABLED` is set to true. 

* Chat with the app

## What to Check
Verify that the following are valid
* Log messages still appear in Application Insights with message "Token Consumption"

image

### CODE REVIEW

**Review by gaurarpit — APPROVED** [review]
LGTM.

```
                              ______________________________________________
                          .-'                     _                        '.
                        .'                       |-'                        |
                      .'                         |                          |
                   _.'               p         _\_/_         p              |
                _.'                  |       .'  |  '.       |              |
           __..'                     |      /    |    \      |              |
     ___..'                         .T\    ======+======    /T.             |
  ;;;\::::                        .' | \  /      |      \  / | '.           |
  ;;;|::::                      .'   |  \/       |       \/  |   '.         |
  ;;;/::::                    .'     |   \       |        \  |     '.       |
        ''.__               .'       |    \      |         \ |       '.     |
             ''._          <_________|_____>_____|__________>|_________>    |
                 '._     (___________|___________|___________|___________)  |
                    '.    \;;;o;;;;o;;;;;o;;;;;o;;;;;o;;;;;o;;;;;o;;;;/   |
                      '.~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~   |
                        '. ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~  |
                          '-.______________________________________________.'
```

File: code/backend/batch/utilities/helpers/EnvHelper.py [hunk 1]
```diff
@@ -164,9 +164,6 @@ def __load_config(self, **kwargs) -> None:
         # APPINSIGHTS_ENABLED will be True when the application runs in App Service
         self.APPINSIGHTS_ENABLED = self.get_env_var_bool("APPINSIGHTS_ENABLED", "False")
 
-        self.APPINSIGHTS_CONNECTION_STRING = os.getenv(
```
[the remark this answers is not in the document]
Comment (liammoat) [reply code/backend/batch/utilities/helpers/EnvHelper.py:167]: It is indeed. From what I understand, it is used as an environment variable implicitly by `configure_azure_monitor` (See [here](https://github.com/Azure/azure-sdk-for-python/tree/dd7fd47bd53a7da76e5e116cef6ccf0b6f58d392/sdk/monitor/azure-monitor-opentelemetry#usage)). For this reason, I kept it in the Bicep.

I removed it here as we don't reference it directly in the code.

What do you think?

[the remark this answers is not in the document]
Comment (ross-p-smith) [reply code/backend/batch/utilities/helpers/EnvHelper.py:167]: The link talks about `APPLICATIONINSIGHTS_CONNECTION_STRING`, whereas we are removing `APPINSIGHTS_CONNECTION_STRING`

**Review by cecheta — DISMISSED** [review]

**Comment by ross-p-smith** [comment]
All great, the only thing I'm worried about is whether we could get ourselves in a state where existing customers, their app service will have the old 'APPINSIGHTS_ENABLED' flag set in the app settings, but the code could be the new version. Are we worried @cecheta ? It's just a debugging thing

**Comment by cecheta** [comment]
> All great, the only thing I'm worried about is whether we could get ourselves in a state where existing customers, their app service will have the old 'APPINSIGHTS_ENABLED' flag set in the app settings, but the code could be the new version. Are we worried @cecheta ? It's just a debugging thing

True, but I don't think this is the worst thing in the world, they just need to update the env var on their end

**Review by cecheta — APPROVED** [review]

**Review by ross-p-smith — APPROVED** [review]
Partridge

**Pull request merged**